src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,91 @@
 package mate.academy.lib;
 
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Class<? extends Annotation> annotationForInjection = Inject.class;
+    private final Class<? extends Annotation> annotationForImplementation = Component.class;
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+
+    {
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+    }
+
+    private Injector() {
+    }
```

Comment:
Why do you need this constructor?

Reply:
This constructor prevent create the others instances of class Injector.
This is the part of pattern Signlton